cache/in-memory/src/event/role.rs

```diff
@@ -151,9 +153,11 @@ mod tests {
             assert!(guild_2_role_ids.iter().all(|id| cached_roles.contains(id)));
 
             // Check for the cached role
-            assert!(guild_2_roles
-                .into_iter()
-                .all(|role| cache.role(role.id).expect("Role missing from cache") == role))
+            assert!(guild_2_roles.into_iter().all(|role| cache
```

Comment:
Same as above

Reply:
No, because a comparison is being done over all the elements of the iterator rather than the iterator itself. This code may need to be refactored but I would prefer to leave that to another PR.